src/datanode/src/config.rs

```diff
@@ -53,6 +53,18 @@ pub enum ObjectStoreConfig {
     Gcs(GcsConfig),
 }
 
+impl ObjectStoreConfig {
+    pub fn extract_variant_name(&self) -> &'static str {
```

Comment:
This has a restriction that we can only specify one store for each kind. Can we add an optional `name` field to the `ObjectStoreConfig` and use that name as the store name?

Reply:
After https://github.com/GreptimeTeam/greptimedb/pull/2733#issuecomment-1815505396 lands, I'll address this.